src/main/java/hudson/plugins/jacoco/JacocoPublisher.java

```diff
@@ -451,13 +461,18 @@ public void perform(@Nonnull Run<?, ?> run, @Nonnull FilePath filePath, @Nonnull
         logger.print("\n[JaCoCo plugin] Saving matched class directories for class-pattern: " + classPattern + ": ");
         for (FilePath file : matchedClassDirs) {
             dir.saveClassesFrom(file);
-            logger.print(" " + file);
```

Comment:
E.g. you can take a look at the build at https://builds.apache.org/view/POI/job/POI-DSL-1.6/, it has thousands of class-files as well, but we only configure the class-directories "build/classes,build/excelant-classes,build/ooxml-classes,build/scratchpad-classes,build/*/build/classes", this results in the following nice and helpful log-output:

```
[JaCoCo plugin] Collecting JaCoCo coverage data...
[JaCoCo plugin] build/*.exec,build/*/build/jacoco/*.exec;build/classes,build/excelant-classes,build/ooxml-classes,build/scratchpad-classes,build/*/build/classes;src/java,src/excelant/java,src/ooxml/java,src/scratchpad/src; locations are configured
```

Configuring it differently so that it needs to iterate over all classes separately might even cause quite an overhead when executing the plugin...